ballerina/utils.bal

```diff
@@ -0,0 +1,239 @@
+
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/url;
+
+type SimpleBasicType string|boolean|int|float|decimal;
+
+# Represents encoding mechanism details.
+type Encoding record {
+    # Defines how multiple values are delimited
+    string style = FORM;
+    # Specifies whether arrays and objects should generate as separate fields
+    boolean explode = true;
+    # Specifies the custom content type
+    string contentType?;
+    # Specifies the custom headers
+    map<any> headers?;
+};
+
+enum EncodingStyle {
+    DEEPOBJECT, FORM, SPACEDELIMITED, PIPEDELIMITED
+}
+
+final Encoding & readonly defaultEncoding = {};
+
+# Serialize the record according to the deepObject style.
+#
+# + parent - Parent record name
+# + anyRecord - Record to be serialized
+# + return - Serialized record as a string
+isolated function getDeepObjectStyleRequest(string parent, record {} anyRecord) returns string {
+    string[] recordArray = [];
+    foreach [string, anydata] [key, value] in anyRecord.entries() {
+        if value is SimpleBasicType {
+            recordArray.push(parent + "[" + key + "]" + "=" + getEncodedUri(value.toString()));
+        } else if value is SimpleBasicType[] {
+            recordArray.push(getSerializedArray(parent + "[" + key + "]" + "[]", value, DEEPOBJECT, true));
+        } else if value is record {} {
+            string nextParent = parent + "[" + key + "]";
+            recordArray.push(getDeepObjectStyleRequest(nextParent, value));
+        } else if value is record {}[] {
+            string nextParent = parent + "[" + key + "]";
+            recordArray.push(getSerializedRecordArray(nextParent, value, DEEPOBJECT));
+        }
+        recordArray.push("&");
+    }
+    _ = recordArray.pop();
+    return string:'join("", ...recordArray);
+}
+
+# Serialize the record according to the form style.
+#
+# + parent - Parent record name
+# + anyRecord - Record to be serialized
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized record as a string
+isolated function getFormStyleRequest(string parent, record {} anyRecord, boolean explode = true) returns string {
+    string[] recordArray = [];
+    if explode {
+        foreach [string, anydata] [key, value] in anyRecord.entries() {
+            if (value is SimpleBasicType) {
```

Comment:
```suggestion
            if value is SimpleBasicType {
```

Reply:
check for other places

Reply:
https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2#discussion_r1533541605